ofctrl/fgraphFlow.go

```diff
@@ -207,161 +214,161 @@ func (self *Flow) Type() string {
-		vidField := openflow13.NewVlanIdField(0, nil)
-		vidField.Value = new(openflow13.VlanIdField)
+		vidField := openflow15.NewVlanIdField(0, nil)
+		vidField.Value = new(openflow15.VlanIdField)
 		ofMatch.AddField(*vidField)
 	} else if self.Match.VlanId != nil {
-		vidField := openflow13.NewVlanIdField(*self.Match.VlanId, self.Match.VlanMask)
+		vidField := openflow15.NewVlanIdField(*self.Match.VlanId, self.Match.VlanMask)
 		ofMatch.AddField(*vidField)
 	}
 
 	// Handle ARP Oper type
 	if self.Match.ArpOper != 0 {
-		arpOperField := openflow13.NewArpOperField(self.Match.ArpOper)
+		arpOperField := openflow15.NewArpOperField(self.Match.ArpOper)
 		ofMatch.AddField(*arpOperField)
 	}
 
 	// Handle ARP THA
 	if self.Match.ArpTha != nil {
-		arpTHAField := openflow13.NewArpThaField(*self.Match.ArpTha)
+		arpTHAField := openflow15.NewArpThaField(*self.Match.ArpTha)
 		ofMatch.AddField(*arpTHAField)
 	}
 
 	// Handle ARP SHA
 	if self.Match.ArpSha != nil {
-		arpSHAField := openflow13.NewArpShaField(*self.Match.ArpSha)
+		arpSHAField := openflow15.NewArpShaField(*self.Match.ArpSha)
 		ofMatch.AddField(*arpSHAField)
 	}
 
 	// Handle ARP TPA
 	if self.Match.ArpTpa != nil {
-		arpTPAField := openflow13.NewArpTpaField(*self.Match.ArpTpa)
+		arpTPAField := openflow15.NewArpTpaField(*self.Match.ArpTpa)
 		ofMatch.AddField(*arpTPAField)
 	}
 
 	// Handle ARP SPA
 	if self.Match.ArpSpa != nil {
-		arpSPAField := openflow13.NewArpSpaField(*self.Match.ArpSpa)
+		arpSPAField := openflow15.NewArpSpaField(*self.Match.ArpSpa)
 		ofMatch.AddField(*arpSPAField)
 	}
 
 	// Handle IP Dst
 	if self.Match.IpDa != nil {
 		if self.Match.IpDa.To4() != nil {
-			ipDaField := openflow13.NewIpv4DstField(*self.Match.IpDa, self.Match.IpDaMask)
+			ipDaField := openflow15.NewIpv4DstField(*self.Match.IpDa, self.Match.IpDaMask)
 			ofMatch.AddField(*ipDaField)
 		} else {
-			ipv6DaField := openflow13.NewIpv6DstField(*self.Match.IpDa, self.Match.IpDaMask)
+			ipv6DaField := openflow15.NewIpv6DstField(*self.Match.IpDa, self.Match.IpDaMask)
 			ofMatch.AddField(*ipv6DaField)
 		}
 	}
 
 	// Handle IP Src
 	if self.Match.IpSa != nil {
 		if self.Match.IpSa.To4() != nil {
-			ipSaField := openflow13.NewIpv4SrcField(*self.Match.IpSa, self.Match.IpSaMask)
+			ipSaField := openflow15.NewIpv4SrcField(*self.Match.IpSa, self.Match.IpSaMask)
 			ofMatch.AddField(*ipSaField)
 		} else {
-			ipv6SaField := openflow13.NewIpv6SrcField(*self.Match.IpSa, self.Match.IpSaMask)
+			ipv6SaField := openflow15.NewIpv6SrcField(*self.Match.IpSa, self.Match.IpSaMask)
 			ofMatch.AddField(*ipv6SaField)
 		}
 	}
 
 	// Handle IP protocol
 	if self.Match.IpProto != 0 {
-		protoField := openflow13.NewIpProtoField(self.Match.IpProto)
+		protoField := openflow15.NewIpProtoField(self.Match.IpProto)
 		ofMatch.AddField(*protoField)
 	}
 
 	// Handle IP dscp
 	if self.Match.IpDscp != 0 {
-		dscpField := openflow13.NewIpDscpField(self.Match.IpDscp)
+		dscpField := openflow15.NewIpDscpField(self.Match.IpDscp, nil)
```

Comment:
This call is conflict with the implementation in libOpenflow, or you forgot to push the update in libOpenflow?

Reply:
Yes I forgot to push libopenflow changes. The changes are now pushed.